.github/workflows/hadolint.yml

```diff
@@ -0,0 +1,20 @@
+on:
+  workflow_call:
+jobs:
+  hadolint:
+    runs-on: ubuntu-latest
+    steps:
+      - 
+        name: Checkout Code
+        uses: actions/checkout@v3
```

Comment:
Is there any reason why this actions/checkout is using v3 version?

Reply:
will upgrade it to v4